File: echo-network-backend/src/main/java/com/wqz/echonetwork/entity/Circle.java
package com.wqz.echonetwork.entity;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

/**
 * 代码不注释，同事两行泪！（给！爷！写！）
 * Elegance is not a dispensable luxury but a quality that decides between success and failure!
 * Created by Wu Qizhen on 2025.10.11
 */
public class Circle { // 圈子

    private Long id; // 圈子 ID

    private String name; // 圈子名称

    private String description; // 圈子描述

    private Long creatorId; // 创建者 ID

    private LocalDateTime createTime; // 创建时间

    /* private Set<Long> memberIds = new HashSet<>(); // 用户成员 ID */

    protected void onCreate() {
        createTime = LocalDateTime.now();
    }
}

File: echo-network-backend/src/main/java/com/wqz/echonetwork/entity/po/Circle.kt
package com.wqz.echonetwork.entity.po

import java.time.LocalDateTime

/**
 * 代码不注释，同事两行泪！（给！爷！写！）
 * Elegance is not a dispensable luxury but a quality that decides between success and failure!
 * Created by Wu Qizhen on 2025.10.11
 */
data class Circle(
    var id: Long? = null, // 圈子 ID
    var name: String = "", // 圈子名称
    var description: String = "", // 圈子描述
    var creatorId: Long? = null, // 创建者 ID
    var createTime: LocalDateTime? = null // 创建时间
) {

    init {
        onCreate()
    }

    fun onCreate() {
        createTime = LocalDateTime.now()
    }
}
